### PR TITLE
[Customer_flow]Customer service repository

### PR DESCRIPTION
**Pull request by Alexander1iubchenko**

### CODE REVIEW

File: src/main/java/com/application/bookingservice/service/customer/CustomerServiceImpl.java [hunk 1]
```diff
@@ -37,17 +42,36 @@ public CustomerRegistrationResponseDto register(CustomerRegistrationRequestDto r
     }
 
     @Override
-    public Object updateRole(Long id, Object updateRequestDto) {
-        return null;
+    public CustomerResponseDtoWithRoles updateRole(Long id,
+                                                   CustomerUpdateRoleRequestDto updateRequestDto) {
+        Customer existingCustomer = customerRepository.findById(id)
+                .orElseThrow(() -> new EntityNotFoundException("Customer is "
+                        + "not found with id: " + id));
```
Comment (BohdanTrue) [inline src/main/java/com/application/bookingservice/service/customer/CustomerServiceImpl.java:49]: make it constant

File: src/main/java/com/application/bookingservice/dto/customer/CustomerDto.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
+package com.application.bookingservice.dto.customer;
+
+import jakarta.validation.constraints.Email;
+import jakarta.validation.constraints.NotNull;
+import lombok.Data;
+
+@Data
+public class CustomerDto {
```
Comment (BohdanTrue) [inline src/main/java/com/application/bookingservice/dto/customer/CustomerDto.java:8]: CustomerRequestDto

File: src/main/java/com/application/bookingservice/dto/customer/CustomerResponseDtoWithRoles.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    @NotBlank(message = "Name cannot be blank")
+    private String email;
+    @Size(max = 50)
+    @NotBlank(message = "Name cannot be blank")
+    private String firstName;
+    @Size(max = 50)
+    @NotBlank(message = "Name cannot be blank")
+    private String lastName;
+    private Set<Role> roles;
+}
```
Comment (BohdanTrue) [inline src/main/java/com/application/bookingservice/dto/customer/CustomerResponseDtoWithRoles.java:22]: are you sure, that validation annotation is necessary for ResponseDto's?

**Review by BohdanTrue — CHANGES_REQUESTED** [review]

File: src/main/java/com/application/bookingservice/dto/customer/CustomerDto.java (hunk 2 above)
Comment (fedorovychh) [reply src/main/java/com/application/bookingservice/dto/customer/CustomerDto.java:8]: agree

File: src/main/java/com/application/bookingservice/dto/customer/CustomerResponseDtoWithRoles.java (hunk 3 above)
Comment (fedorovychh) [reply src/main/java/com/application/bookingservice/dto/customer/CustomerResponseDtoWithRoles.java:22]: agree

File: src/main/java/com/application/bookingservice/dto/customer/CustomerDto.java [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+    @Email
+    @NotNull(message = "can't be null")
```
Comment (fedorovychh) [inline src/main/java/com/application/bookingservice/dto/customer/CustomerDto.java:10]: We validate String as NotBlank, another types as NotNull

File: src/main/java/com/application/bookingservice/dto/customer/CustomerDto.java (hunk 2 above)
Comment (fedorovychh) [reply src/main/java/com/application/bookingservice/dto/customer/CustomerDto.java:8]: And than make separately dto for  response. Cause you have a method where you update customer and get in param and return this CustomerDto. So you can make CustomerUpdateRequestDto and CustomerUpdateResponseDto. They will have the same params - OK

File: src/main/java/com/application/bookingservice/mapper/CustomerMapper.java [hunk 5]
```diff
@@ -11,4 +13,9 @@ public interface CustomerMapper {
     CustomerRegistrationResponseDto toDto(Customer customer);
 
     Customer toModel(CustomerRegistrationRequestDto requestDto);
+
+    CustomerDto toCustomerDto(Customer customer);
```
Comment (fedorovychh) [inline src/main/java/com/application/bookingservice/mapper/CustomerMapper.java:17]: toUpdateDto

File: src/main/java/com/application/bookingservice/mapper/CustomerMapper.java [hunk 6]
```diff
@@ -11,4 +13,9 @@ public interface CustomerMapper {
...
+
+    CustomerResponseDtoWithRoles toCustomerResponseDto(Customer customer);
```
Comment (fedorovychh) [inline src/main/java/com/application/bookingservice/mapper/CustomerMapper.java:19]: toDtoWithRoles

File: src/main/java/com/application/bookingservice/service/customer/CustomerServiceImpl.java [hunk 7]
```diff
@@ -37,17 +42,36 @@ public CustomerRegistrationResponseDto register(CustomerRegistrationRequestDto r
...
+        return customerMapper.toCustomerResponseDto(customerRepository.save(existingCustomer));
     }
 
     @Override
-    public Object getById(Long customerId) {
-        return null;
+    public CustomerDto getById(Long customerId) {
+        Customer customer = customerRepository
+                .findById(customerId)
+                .orElseThrow(() -> new EntityNotFoundException("Customer is not found with id: "
```
Comment (fedorovychh) [inline src/main/java/com/application/bookingservice/service/customer/CustomerServiceImpl.java:61]: In this project we make messages as constants, so fix all please

**Review by fedorovychh — APPROVED** [review]
LGTM

**Review by nshtykh — APPROVED** [review]

**Review by BohdanTrue — APPROVED** [review]

**Pull request merged**